Support `gf` for finding components instantiated via angle-brackets.

saravanak opened the pull request:
@AndrewRadev , thanks for the wonderful plugin, I've been using it for some time, and love it. 

This PR scratches my own itch: to support `gf` for angle bracketed component instantiation. 

I've added a couple of specs and adhered the conventions. Please let me know if this is good enuf.

saravanak committed:
Support `gf` for finding components instantiated via angle-brackets.

saravanak committed:
Reduce duplication.

saravanak commented:
Not sure why travis is failing, but the tests passes in dev..ran it a couple times..

AndrewRadev commented on autoload/ember_tools/gf.vim:
```diff
@@ -120,26 +120,51 @@ function! ember_tools#gf#Model()
+  set iskeyword-=/
+  let angle_bracketed_component_name = expand('<cword>')
+  set iskeyword-=:
+  set iskeyword+=/
```
I'd like to be more precise in changing the `iskeyword` param -- what worries me in this case is that we're adding and removing properties, so if `:` is in `iskeyword` due to a user setting, it would be removed after this line. I'd like to try this:

``` vim
try
  let saved_iskeyword = &iskeyword
  set iskeyword += ...
  let angle_bracketed_...
finally
  let &iskeyword = saved_iskeyword
endtry
```

I see that in the codebase I also do what you did, so it's understandable you used the same pattern -- after the PR is merged, I'll go around and I'll apply it elsewhere too.

AndrewRadev commented on autoload/ember_tools/gf.vim:
```diff
@@ -120,26 +120,51 @@ function! ember_tools#gf#Model()
 
-function! ember_tools#gf#TemplateComponent()
+"Should be a 'dasherized' name: `foo-bar/bazz`
+function! ember_tools#gf#FindComponent(component_name)
```
I try to keep a minimal public interface to make it clear which functions are used elsewhere and which aren't. This function is basically a sort of common helper function, so I would use the `s:` form to keep it script-local, so essentially private. I would name it `s:FindComponent`. Maybe it could be located at the bottom of the file with the other functions it uses, so:

``` vim
function! s:FindComponent(component_name)
function! s:FindComponentLogic(component_name)
function! s:FindComponentTemplate(component_name)
```

Does that make sense?

saravanak replied:
I missed the script-local part. In the process, renamed the function to a more descriptive `s:FindDasherizedComponent`.

AndrewRadev reviewed (changes requested):
Thank you very much for your work! I have "angle-bracket component" support in my own note files, so I'm very happy to see a PR for it :).

I've left some notes about code changes -- let me know what you think. Don't worry about the build failure, from what I can see, it seems to be a random one -- rerunning made it green. (It makes me sad there's a randomly-failing test, but I don't have an idea why it could be happening)

saravanak commented:
Sure, I will make these changes and resubmit the PR! :+1:

saravanak committed:
Address review comments:

 - Make global FindComponent as s:FindDasherizedComponent
 - Use try/catch/finally to rollback `iskeyword` changes.

saravanak committed:
Move up the new function to the suggested order.

AndrewRadev commented:
Thanks, this works -- I'll go ahead and merge it now.
